hasher-matcher-actioner/webapp/src/Upload.js

```diff
@@ -7,41 +7,57 @@ import { Link } from 'react-router-dom';
 import { Button, Col, Collapse, Image, Row } from 'react-bootstrap';
 import Spinner from 'react-bootstrap/Spinner';
 
+import { postAPI } from './Api'
+
 export default function Upload() {
-  const [ fileName, setFileName ] = useState('Select Browse to choose a file.');
-  const [ submitting, setSubmitting ] = useState(false);
-  const [ submitted, setSubmitted ] = useState(false);
+  const [fileName, setFileName] = useState('Select Browse to choose a file.');
+  const [submitting, setSubmitting] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
+  const [image, setImage] = useState({ preview: "", raw: "" });
+
+  function handleUpload() {
+    const formData = new FormData();
+    formData.append("image", image.raw);
+    postAPI('upload', formData).then((response) => {
+      // TODO check for succesfull upload
+      setSubmitted(true);
+      setSubmitting(false)
+    })
+  }
+
   return (
     <>
       <h1>Upload</h1>
       <Row className="mt-3">
         <Col md={6}>
           <Collapse in={!submitting && !submitted}>
             <div>
-            <p>
-              Browse then select a file to upload for hashing, matching and actioning, then select <strong>Submit</strong>.
+              <p>
+                Browse then select a file to upload for hashing, matching and actioning, then select <strong>Submit</strong>.
             </p>
-            <div className="custom-file">
-              <input type="file" className="custom-file-input" id="customFile" onChange={(e) => {
-                setFileName(e.nativeEvent.path[0].files[0].name);
-              }} />
-              <label className="custom-file-label" for="customFile">{fileName}</label>
-            </div>
-            <div className="mt-3">
-              <Button onClick={() => {
-                setSubmitting(true);
-                setTimeout(() => {
-                  setSubmitted(true);
-                  setSubmitting(false);
-                }, 3000);
-              }}>Submit</Button>
-            </div>
+              <div className="custom-file">
+                <input type="file" className="custom-file-input" id="customFile" onChange={(e) => {
+                  setFileName(e.nativeEvent.path[0].files[0].name);
+                  setImage({
+                    preview: URL.createObjectURL(e.nativeEvent.path[0].files[0]),
+                    raw: e.nativeEvent.path[0].files[0]
```

Comment:
Mini nit / preference: dereference `e.nativeEvent.path[0].files[0]` to something like `file` and then use `file` at 40, 42 and 43.